### ISSUE TITLE
Training fine-tuning inside reasoning, if you set the number of slices (max_chars)?

### ISSUE THREAD

**Issue by ylf2024** [SWivid/F5-TTS#888]
### Checks

- [x] This template is only for research question, not usage problems, feature requests or bug reports.
- [x] I have thoroughly reviewed the project documentation and read the related paper(s).
- [x] I have searched for existing issues, including closed ones, no similar questions.
- [x] I am using English to submit this issue to facilitate community communication.

### Question details

Training fine-tuning inside reasoning, if you set the number of slices (max_chars)?thanks

**Comment by SWivid** [comment]
not set, just for simple test

**Comment by ylf2024** [comment]
I am confused as to why the new version lacks so much detail in tone compared to the previous version. The new version is very AI when it is synthesized, without any emotion or tone. The cadence is very good compared to the old version in terms of tone pauses. The only downside is the vibrato. I am currently using the November 2024 version.

The same paragraph of text: the new version is generated at 18s, the old version is generated at 22s, and this 4-second difference is in the tone pause. How do I solve the vibrato problem if I keep using the old version? How can the new version have the same effect as the old version with tone pause?

[issue.zip](https://github.com/user-attachments/files/19368821/issue.zip)

**Comment by SWivid** [comment]
The new version will mimick more thoroughly the ref_audio, so if ref_audio's tone is very flat, the gen_audio will also be flat.
I dunno how is the ref_audio you used here, is it flat or has certain prosody?

**Comment by ylf2024** [comment]
Hello thank you for your reply, in fact, the problem with the new version is too fast, for example, I gave screenshots, reference sounds and generated speech in comma understanding is insufficient, my reference is due to emotion. The resulting sound is fast, resulting in inadequate processing in detail.

![Image](https://github.com/user-attachments/assets/d79fcc8e-a42c-4f44-860f-09ad538b7a24)

**Comment by SWivid** [comment]
would it be possible that you provide the reference audio here that we could test on our side?

**Comment by ylf2024** [comment]
What's your email address?

**Comment by SWivid** [comment]
swivid@sjtu.edu.cn

**Comment by ylf2024** [comment]
It has been sent to you. Please note that

**Comment by SWivid** [comment]
Got the reason of your issue, try pull the latest code.
In your version (that is, after v1 update but before v1.0.5 fix), -1 is actually not for random but a fixed seed.

You can see that I just test with infer gradio, that different seed can offer better result.
And if you pull latest code for finetune gradio, will also enable correct random seed to test model
![Image](https://github.com/user-attachments/assets/ba2affef-9301-4b13-a102-ccd8ac9015b0)

**Comment by ylf2024** [comment]
NONO&nbsp;一样如此，需要多次抽卡才能获得，这样是否合理？并且通过抽卡获取的种子，是否满足其他的文本需求，这无疑是增加了更多的不确定性。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "SWivid/F5-TTS"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年3月21日(星期五) 上午10:02
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [SWivid/F5-TTS] Training fine-tuning inside reasoning, if you set the number of slices (max_chars)? (Issue #888)



   
Got the reason of your issue, try pull the latest code.
 In your version (that is, after v1 update but before v1.0.5 fix), -1 is actually not for random but a fixed seed.
 
You can see that I just test with infer gradio, that different seed can offer better result.
 And if you pull latest code for finetune gradio, will also enable correct random seed to test model
 image.png (view on web)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  SWivid left a comment (SWivid/F5-TTS#888)
 
Got the reason of your issue, try pull the latest code.
 In your version (that is, after v1 update but before v1.0.5 fix), -1 is actually not for random but a fixed seed.
 
You can see that I just test with infer gradio, that different seed can offer better result.
 And if you pull latest code for finetune gradio, will also enable correct random seed to test model
 image.png (view on web)

**Comment by SWivid** [comment]
所以我们并没有强制要求你使用v1版本，并且为v0版本提供了完全的兼容
无论是微调还是推理（在custom里填一下就行）的gradio，你都可以使用之前的版本

如果只是为了稳定性，你完全可以自己训一个模型
同时我们也很欢迎你提供一个满足所有人需求的，兼顾稳定性、零样本克隆能力、特定说话人效果的base模型~

**Comment by ylf2024** [comment]
The same is true, it takes many times to draw cards to get it. Is this reasonable? And whether the seeds obtained by card drawing meet other text requirements undoubtedly increases more uncertainty.


I sent you two paragraphs through the latest version 1.06, and last year.




------------------ 原始邮件 ------------------
发件人:                                                                                                                        "SWivid/F5-TTS"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年3月21日(星期五) 上午10:02
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [SWivid/F5-TTS] Training fine-tuning inside reasoning, if you set the number of slices (max_chars)? (Issue #888)



   
Got the reason of your issue, try pull the latest code.
 In your version (that is, after v1 update but before v1.0.5 fix), -1 is actually not for random but a fixed seed.
 
You can see that I just test with infer gradio, that different seed can offer better result.
 And if you pull latest code for finetune gradio, will also enable correct random seed to test model
 image.png (view on web)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  SWivid left a comment (SWivid/F5-TTS#888)
 
Got the reason of your issue, try pull the latest code.
 In your version (that is, after v1 update but before v1.0.5 fix), -1 is actually not for random but a fixed seed.
 
You can see that I just test with infer gradio, that different seed can offer better result.
 And if you pull latest code for finetune gradio, will also enable correct random seed to test model
 image.png (view on web)

**Comment by ylf2024** [comment]
我目前遇到的困难就是在用老的版本，遇到颤音的问题，生成的语音不稳定，也就是您之前说到的，每个切片在尾部的几个发音会发虚。
目前我在新版本上遇到的问题也有其他的人反馈过就是，语音速度较快。在 逗号和句号处理上，比较不稳定。我给您发的也只是一段话而已，如果在长文本处理时，更加的显现生成的语音急促的问题。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "SWivid/F5-TTS"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年3月21日(星期五) 上午10:09
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [SWivid/F5-TTS] Training fine-tuning inside reasoning, if you set the number of slices (max_chars)? (Issue #888)



   
所以我们并没有强制要求你使用v1版本，并且为v0版本提供了完全的兼容
 无论是微调还是推理（在custom里填一下就行）的gradio，你都可以使用之前的版本
 
如果只是为了稳定性，你完全可以自己训一个模型
 同时我们也很欢迎你提供一个满足所有人需求的，兼顾稳定性、零样本克隆能力、特定说话人效果的base模型~

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  SWivid left a comment (SWivid/F5-TTS#888)
 
所以我们并没有强制要求你使用v1版本，并且为v0版本提供了完全的兼容
 无论是微调还是推理（在custom里填一下就行）的gradio，你都可以使用之前的版本
 
如果只是为了稳定性，你完全可以自己训一个模型
 同时我们也很欢迎你提供一个满足所有人需求的，兼顾稳定性、零样本克隆能力、特定说话人效果的base模型~

**Comment by SWivid** [comment]
> 我目前遇到的困难就是在用老的版本，遇到颤音的问题，生成的语音不稳定，也就是您之前说到的，每个切片在尾部的几个发音会发虚。

这个之前也解释了，尾部电音是因为长度超过训练集常见的音频长度。v0版本主要问题是欠normalize，爆破音有刺啦声。

> 目前我在新版本上遇到的问题也有其他的人反馈过就是，语音速度较快。在 逗号和句号处理上，比较不稳定。我给您发的也只是一段话而已，如果在长文本处理时，更加的显现生成的语音急促的问题。

这个和上面的原因是一致的，因为v1模型对齐建模更强，所以对长度的敏感性更高。对于超过训练集常见长度的长音频外推能力不如v0。
一旦模型遇到了超过它舒适区的（训练集学得好的）的长度的推理，就会尝试读得快来把内容塞到舒适区的长度里，这个底层逻辑v0 v1是一致的，而v1你感受到更明显就是因为外推的问题。
治本的方式是：一、用长数据集训练或微调；二、更好的模型结构设计来支持外推。且无论哪种我们都有在更新迭代。

最后还是得说明下，这个项目代码完全都开源的，模型权重也是按照数据集的最低要求开的，所以任何人都能直接应用开发。
虽然我们对社区一直有超过绝大多数项目的友好支持，但希望你明白每个人的精力都有限，同时我们并非盈利所以从逻辑上并不是有直接义务在确保产品上的适配。
相比之下，我们更乐于看到社区共同帮助建设，事实上有许多人都这么做的，我们也很感谢他们。
反馈意见同样宝贵，但是在此之上我们不可能像商业公司一样为所有人提供实时产品售后，只可能支持长期迭代。

**Comment by ylf2024** [comment]
好的明白了感谢您的回复，您也别太过在意，我给您反馈主要也是处于对产品优化的一种帮助，如果您觉得这给您增加了麻烦，我感到很抱歉。谢谢。
